tests/error_message.rs

```diff
@@ -0,0 +1,17 @@
+extern crate snafu;
+
+use std::error::Error as _;
+use snafu::{Snafu, ResultExt};
+
+
+#[derive(Debug, Snafu)]
+enum Error {
+    BadId{ source: std::io::Error },
+}
+
+#[test]
+fn test_error_message() {
+    let err : Error = std::fs::read("/non/existent/file--test").eager_context(BadId{}).unwrap_err();
+    let err_description = format!("{}", err);
+    assert_eq!(err_description, String::from("BadId: No such file or directory (os error 2)"));
```

Comment:
You shouldn't need to allocate the `String` to test against; the string literal is fine.